File: 845C.cpp
#include <cstdio>
#include <cstring>
#include <vector>
#include <iostream>
#include <cmath>
#include <algorithm>
using namespace std;
typedef long long ll;

int main(){
	std::ios::sync_with_stdio(false);
	cin.tie(nullptr); 
	cout.tie(nullptr);

	int n;
	cin >> n;
	vector<pair<int, int>> a(n);
	for (auto& p : a) {
		cin >> p.first >> p.second;
	}

	std::sort(a.begin(), a.end());

	int end1 = -1, end2 = -2;
	for (auto&p : a) {
		if (p.first <= end1 && p.first <= end2) {
			cout << "NO";
			return 0;
		}
		else if (p.first > end1) {
			end1 = p.second;
		} else {
			end2 = p.second;
		}
	}
	cout << "YES";
	return 0;
}

File: 845C.go
package main

import (
	"bufio"
	. "fmt"
	"os"
	"sort"
)

func main() {
	var n int
	in := bufio.NewReader(os.Stdin)
	Fscan(in, &n)

	type pair struct {
		x, y int
	}
	a := make([]pair, n)
	for i := range a {
		Fscan(in, &a[i].x, &a[i].y)
	}
	sort.Slice(a, func(i, j int) bool {
		if a[i].x == a[j].x {
			return a[i].y < a[j].y
		}
		return a[i].x < a[j].x
 	})

	end1, end2 := -1, -1
	for _, p := range a {
		if p.x <= end1 && p.x <= end2 {
			Println("NO")
			return
		} else if p.x > end1 {
			end1 = p.y
		} else {
			end2 = p.y
		}
	}
	Println("YES")
	return
}
